From 9bc21ee7da5de6b2531d47d1af4d7b0a169990b9 Mon Sep 17 00:00:00 2001
From: Jonathan Marten <jjm@keelhaul.me.uk>
Date: Thu, 22 Jan 2015 18:41:19 +0000
Subject: [PATCH 06/24] Allow SYSTEMD_SYSTEM_UNIT_DIR to be overidden

From GitHub issue #343
---
 CMakeLists.txt | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d6e0a9..61230a4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,8 +109,11 @@ if(SYSTEMD_FOUND)
         endif()
     endif()
 
-    execute_process(COMMAND ${PKG_CONFIG_EXECUTABLE} --variable=systemdsystemunitdir systemd OUTPUT_VARIABLE SYSTEMD_SYSTEM_UNIT_DIR)
-    string(REGEX REPLACE "[ \t\n]+" "" SYSTEMD_SYSTEM_UNIT_DIR ${SYSTEMD_SYSTEM_UNIT_DIR})
+    if (NOT DEFINED SYSTEMD_SYSTEM_UNIT_DIR)
+        execute_process(COMMAND ${PKG_CONFIG_EXECUTABLE} --variable=systemdsystemunitdir systemd OUTPUT_VARIABLE SYSTEMD_SYSTEM_UNIT_DIR)
+        string(REGEX REPLACE "[ \t\n]+" "" SYSTEMD_SYSTEM_UNIT_DIR ${SYSTEMD_SYSTEM_UNIT_DIR})
+    endif()
+
     set(MINIMUM_VT 1)
     set(HALT_COMMAND "/usr/bin/systemctl poweroff")
     set(REBOOT_COMMAND "/usr/bin/systemctl reboot")
-- 
1.9.0

