From 7a5a36158ff54371d0ff020cf09e7646007df987 Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Sat, 31 Jan 2015 12:58:45 +0100
Subject: [PATCH 09/24] ChangeLog: Remove empty line

Conform to other release entries.
---
 ChangeLog | 1 -
 1 file changed, 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 165b662..8d55375 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -19,7 +19,6 @@
 
 ## 0.10.0 - 2014-10-16
 ----------------------
-
     + Set default icon theme from greeter theme configuration.
     + Set cursor theme according to greeter theme configuration.
     - Never permit a login as the sddm user (CVE-2014-7271)
-- 
1.9.0

