From 43a726436fed6ed94d7d3e9afa0ff773bc6cf2b2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tomasz=20Pawe=C5=82=20Gajc?= <tpgxyz@gmail.com>
Date: Tue, 18 May 2021 13:11:56 +0200
Subject: [PATCH] sddm service is a part of graphical target

---
 services/sddm.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/sddm.service.in b/services/sddm.service.in
index 03356859..d7870199 100644
--- a/services/sddm.service.in
+++ b/services/sddm.service.in
@@ -3,6 +3,7 @@ Description=Simple Desktop Display Manager
 Documentation=man:sddm(1) man:sddm.conf(5)
 Conflicts=getty@tty1.service
 After=systemd-user-sessions.service getty@tty1.service plymouth-quit.service systemd-logind.service
+PartOf=graphical.target
 StartLimitIntervalSec=30
 StartLimitBurst=2
 
