From f6d4444fadeef14ab7ffc849acdfa5b7066bd8a3 Mon Sep 17 00:00:00 2001
From: leuqarte <leuqarte@users.noreply.github.com>
Date: Wed, 18 Feb 2015 09:14:36 -0800
Subject: [PATCH 17/24] cleanup

---
 data/man/sddm.rst.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/data/man/sddm.rst.in b/data/man/sddm.rst.in
index 4fd3575..82d35c2 100644
--- a/data/man/sddm.rst.in
+++ b/data/man/sddm.rst.in
@@ -25,9 +25,9 @@ Using QtQuick and QML, designers are given the ability to
 easily create pleasant, modern looking interfaces for sddm.
 
 sddm runs the greeter as a system user named **sddm**
-whose home directory need to be set to **/var/lib/sddm**.
+whose home directory needs to be set to **/var/lib/sddm**.
 
-If pam and systemd are available, the greeter will go through logind
+If pam and systemd are available, the greeter will go through logind,
 which will give it access to drm devices.
 
 Distributions without pam and systemd will need to put the **sddm** user
-- 
1.9.0

