From 4bd8647de3ac7edb78c6d6dfbd376909974be6b0 Mon Sep 17 00:00:00 2001
From: Alexander Potashev <aspotashev@gmail.com>
Date: Thu, 19 Mar 2015 04:42:01 +0300
Subject: [PATCH 22/24] Improve Russian translation

---
 data/translations/ru.ts | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/data/translations/ru.ts b/data/translations/ru.ts
index e25c189..433d7b5 100644
--- a/data/translations/ru.ts
+++ b/data/translations/ru.ts
@@ -1,15 +1,15 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE TS>
-<TS version="2.0" language="ru">
+<TS version="2.1" language="ru">
 <context>
     <name>TextConstants</name>
     <message>
         <source>Welcome to %1</source>
-        <translation>Добро пожаловать в %1</translation>
+        <translation>Добро пожаловать на %1</translation>
     </message>
     <message>
         <source>Warning, Caps Lock is ON!</source>
-        <translation>Включен Caps Lock!</translation>
+        <translation>Внимание: нажата клавиша Caps Lock</translation>
     </message>
     <message>
         <source>Layout</source>
@@ -17,11 +17,11 @@
     </message>
     <message>
         <source>Login</source>
-        <translation>Вход</translation>
+        <translation>Войти</translation>
     </message>
     <message>
         <source>Login failed</source>
-        <translation>Вход не удался</translation>
+        <translation>Не удалось войти в систему</translation>
     </message>
     <message>
         <source>Login succeeded</source>
@@ -37,15 +37,15 @@
     </message>
     <message>
         <source>Reboot</source>
-        <translation>Перезагрузка</translation>
+        <translation>Перезагрузить</translation>
     </message>
     <message>
         <source>Session</source>
-        <translation>Сессия</translation>
+        <translation>Сеанс</translation>
     </message>
     <message>
         <source>Shutdown</source>
-        <translation>Выключение</translation>
+        <translation>Выключить</translation>
     </message>
     <message>
         <source>User name</source>
-- 
1.9.0

