From 4d503236df82a7c6cf4ea61f1c3575642466fb16 Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Tue, 31 Mar 2015 22:40:53 +0200
Subject: [PATCH 23/24] Add russian translation improvements to ChangeLog

---
 ChangeLog | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ChangeLog b/ChangeLog
index 8d55375..17a39b6 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -7,6 +7,7 @@
 ## 0.12.0 - XXXX-XX-XX
 ----------------------
     * Add Arabic translation by Safa Alfulaij.
+    * Russian translation improvements by Alexander Potashev.
 
 ## 0.11.0 - 2014-11-20
 ----------------------
-- 
1.9.0

