diff -up sddm-develop/src/greeter/GreeterApp.cpp.3~ sddm-develop/src/greeter/GreeterApp.cpp
--- sddm-develop/src/greeter/GreeterApp.cpp.3~	2023-06-04 16:18:27.196164802 +0200
+++ sddm-develop/src/greeter/GreeterApp.cpp	2023-06-04 16:19:35.168745249 +0200
@@ -305,6 +305,9 @@ namespace SDDM {
 
 int main(int argc, char **argv)
 {
+    // Qt internally may load the xdg portal system early on, prevent this, we do not have a functional session running.
+    qputenv("QT_NO_XDG_DESKTOP_PORTAL", "1");
+
     bool testMode = false;
     // We set an attribute based on the platform we run on.
     // We only know the platform after we constructed QGuiApplication
