From 942e173cc29bfcfec74ad24cff997f76aaaf8f53 Mon Sep 17 00:00:00 2001
From: rparkins999 <aleph0hpela-bughit@yahoo.co.uk>
Date: Tue, 15 Oct 2024 09:10:51 +0100
Subject: [PATCH] Fix for issue  Suspicious code in PamHandle::end #1990

---
 src/helper/backend/PamHandle.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/helper/backend/PamHandle.cpp b/src/helper/backend/PamHandle.cpp
index e8e1d41f8..4493c76e3 100644
--- a/src/helper/backend/PamHandle.cpp
+++ b/src/helper/backend/PamHandle.cpp
@@ -162,7 +162,7 @@ namespace SDDM {
     bool PamHandle::end(int flags) {
         if (!m_handle)
             return false;
-        m_result = pam_end(m_handle, m_result | flags);
+        m_result = pam_end(m_handle, m_silent | flags);
         if (m_result != PAM_SUCCESS) {
             qWarning() << "[PAM] end:" << pam_strerror(m_handle, m_result);
             return false;
