From 9e6a9c12074a7b5c4db31aa409ded2c7ef436892 Mon Sep 17 00:00:00 2001
From: snowcrash79 <35747241+snowcrash79@users.noreply.github.com>
Date: Thu, 23 Jan 2025 17:11:34 -0300
Subject: [PATCH] Updated spanish translation

---
 data/translations/es.ts | 142 ++++++++++++++++++++--------------------
 1 file changed, 71 insertions(+), 71 deletions(-)

diff --git a/data/translations/es.ts b/data/translations/es.ts
index cea361eff..5e042c0cc 100644
--- a/data/translations/es.ts
+++ b/data/translations/es.ts
@@ -1,73 +1,73 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version='1.0' encoding='utf-8'?>
 <!DOCTYPE TS>
-<TS version="2.1" language="es">
-<context>
-    <name>PictureBox</name>
-    <message>
-        <source>Press to login</source>
-        <translation type="unfinished"></translation>
-    </message>
-</context>
-<context>
-    <name>QObject</name>
-    <message>
-        <source>%1 (Wayland)</source>
-        <translation type="unfinished"></translation>
-    </message>
-</context>
-<context>
-    <name>TextConstants</name>
-    <message>
-        <source>Welcome to %1</source>
-        <translation>Bienvenido a %1</translation>
-    </message>
-    <message>
-        <source>Warning, Caps Lock is ON!</source>
-        <translation>¡Atención, el bloqueo de mayúsculas está activado!</translation>
-    </message>
-    <message>
-        <source>Layout</source>
-        <translation>Layout</translation>
-    </message>
-    <message>
-        <source>Login</source>
-        <translation>Iniciar sesión</translation>
-    </message>
-    <message>
-        <source>Login failed</source>
-        <translation>Inicio de sesión fallido</translation>
-    </message>
-    <message>
-        <source>Login succeeded</source>
-        <translation>Inicio de sesión correcto</translation>
-    </message>
-    <message>
-        <source>Password</source>
-        <translation>Contraseña</translation>
-    </message>
-    <message>
-        <source>Enter your username and password</source>
-        <translation>Ingrese su usuario y contraseña</translation>
-    </message>
-    <message>
-        <source>Reboot</source>
-        <translation>Reiniciar</translation>
-    </message>
-    <message>
-        <source>Session</source>
-        <translation>Sesión</translation>
-    </message>
-    <message>
-        <source>Shutdown</source>
-        <translation>Apagar</translation>
-    </message>
-    <message>
-        <source>Username</source>
-        <translation>Usuario</translation>
-    </message>
-    <message>
-        <source>Select your user and enter password</source>
-        <translation>Seleccione su usuario e ingrese su contraseña</translation>
-    </message>
-</context>
+<TS language="es" version="2.1">
+    <context>
+        <name>PictureBox</name>
+        <message>
+            <source>Press to login</source>
+            <translation>Presione para iniciar sesión</translation>
+        </message>
+    </context>
+    <context>
+        <name>QObject</name>
+        <message>
+            <source>%1 (Wayland)</source>
+            <translation>%1 (Wayland)</translation>
+        </message>
+    </context>
+    <context>
+        <name>TextConstants</name>
+        <message>
+            <source>Welcome to %1</source>
+            <translation>Bienvenido(a) a %1</translation>
+        </message>
+        <message>
+            <source>Warning, Caps Lock is ON!</source>
+            <translation>¡Atención, el bloqueo de mayúsculas está activado!</translation>
+        </message>
+        <message>
+            <source>Layout</source>
+            <translation>Diseño</translation>
+        </message>
+        <message>
+            <source>Login</source>
+            <translation>Iniciar sesión</translation>
+        </message>
+        <message>
+            <source>Login failed</source>
+            <translation>Inicio de sesión fallido</translation>
+        </message>
+        <message>
+            <source>Login succeeded</source>
+            <translation>Inicio de sesión correcto</translation>
+        </message>
+        <message>
+            <source>Password</source>
+            <translation>Contraseña</translation>
+        </message>
+        <message>
+            <source>Enter your username and password</source>
+            <translation>Ingrese su usuario y contraseña</translation>
+        </message>
+        <message>
+            <source>Reboot</source>
+            <translation>Reiniciar</translation>
+        </message>
+        <message>
+            <source>Session</source>
+            <translation>Sesión</translation>
+        </message>
+        <message>
+            <source>Shutdown</source>
+            <translation>Apagar</translation>
+        </message>
+        <message>
+            <source>Username</source>
+            <translation>Usuario</translation>
+        </message>
+        <message>
+            <source>Select your user and enter password</source>
+            <translation>Seleccione su usuario e ingrese la contraseña</translation>
+        </message>
+    </context>
 </TS>
