From abb9f9eea9ffb31506a89d5f5de1060bea076056 Mon Sep 17 00:00:00 2001
From: nerdopolis <bluescreen_avenger@verizon.net>
Date: Tue, 20 Feb 2024 21:45:18 -0500
Subject: [PATCH] UserSession: Only act on the VT when XDG_VTNR is set

---
 src/helper/UserSession.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/helper/UserSession.cpp b/src/helper/UserSession.cpp
index 860f16547..03fd396ee 100644
--- a/src/helper/UserSession.cpp
+++ b/src/helper/UserSession.cpp
@@ -202,7 +202,7 @@ namespace SDDM {
             // when this is true we'll take control of the tty
             bool takeControl = false;
 
-            if (vtFd > 0) {
+            if (vtNumber > 0 && vtFd > 0) {
                 dup2(vtFd, STDIN_FILENO);
                 ::close(vtFd);
                 takeControl = true;
@@ -228,7 +228,8 @@ namespace SDDM {
                 }
             }
 
-            VirtualTerminal::jumpToVt(vtNumber, x11UserSession);
+            if (vtNumber > 0)
+                VirtualTerminal::jumpToVt(vtNumber, x11UserSession);
         }
 
 #ifdef Q_OS_LINUX
