From bc9eee8280275723767213220e88f6b14157ba1f Mon Sep 17 00:00:00 2001
From: Sergiu Bivol <sergiu@cip.md>
Date: Thu, 5 Mar 2026 09:22:02 +0000
Subject: [PATCH] Update Romanian translation

---
 data/translations/ro.ts | 212 ++++++++++++++++++++++++++--------------
 1 file changed, 141 insertions(+), 71 deletions(-)

diff --git a/data/translations/ro.ts b/data/translations/ro.ts
index 062a43ed4..ca0cc1c6c 100644
--- a/data/translations/ro.ts
+++ b/data/translations/ro.ts
@@ -1,73 +1,143 @@
-<?xml version="1.0" encoding="utf-8"?>
+<?xml version='1.0' encoding='utf-8'?>
 <!DOCTYPE TS>
-<TS version="2.1" language="ro">
-<context>
-    <name>PictureBox</name>
-    <message>
-        <source>Press to login</source>
-        <translation type="unfinished"></translation>
-    </message>
-</context>
-<context>
-    <name>QObject</name>
-    <message>
-        <source>%1 (Wayland)</source>
-        <translation type="unfinished"></translation>
-    </message>
-</context>
-<context>
-    <name>TextConstants</name>
-    <message>
-        <source>Welcome to %1</source>
-        <translation>Bun venit la %1</translation>
-    </message>
-    <message>
-        <source>Warning, Caps Lock is ON!</source>
-        <translation>Atenție, tasta Caps Lock este apăsată!</translation>
-    </message>
-    <message>
-        <source>Layout</source>
-        <translation>Aranjament</translation>
-    </message>
-    <message>
-        <source>Login</source>
-        <translation>Autentificare</translation>
-    </message>
-    <message>
-        <source>Login failed</source>
-        <translation>Autentificare eșuată</translation>
-    </message>
-    <message>
-        <source>Login succeeded</source>
-        <translation>Autentificare cu succes</translation>
-    </message>
-    <message>
-        <source>Password</source>
-        <translation>Parolă</translation>
-    </message>
-    <message>
-        <source>Enter your username and password</source>
-        <translation>Introdu numele de utilizator și parola</translation>
-    </message>
-    <message>
-        <source>Reboot</source>
-        <translation>Repornește</translation>
-    </message>
-    <message>
-        <source>Session</source>
-        <translation>Sesiune</translation>
-    </message>
-    <message>
-        <source>Shutdown</source>
-        <translation>Oprește</translation>
-    </message>
-    <message>
-        <source>Username</source>
-        <translation>Nume de utilizator</translation>
-    </message>
-    <message>
-        <source>Select your user and enter password</source>
-        <translation>Alege numele tău și introdu parola</translation>
-    </message>
-</context>
+<TS language="ro" version="2.1">
+    <context>
+        <name>PictureBox</name>
+        <message>
+            <location filename="../../components/2.0/PictureBox.qml" line="125"/>
+            <source>Press to login</source>
+            <translation>Apăsați pentru autentificare</translation>
+        </message>
+    </context>
+    <context>
+        <name>SDDM::SessionModel</name>
+        <message>
+            <location filename="../../src/greeter/SessionModel.cpp" line="113"/>
+            <source>%1 (Wayland)</source>
+            <translation>%1 (Wayland)</translation>
+        </message>
+    </context>
+    <context>
+        <name>TextConstants</name>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="49"/>
+            <source>Welcome to %1</source>
+            <translation>Bun venit la %1</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="28"/>
+            <source>Warning, Caps Lock is ON!</source>
+            <translation>Atenție, Caps Lock e APĂSAT!</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="29"/>
+            <source>Layout</source>
+            <translation>Aranjament</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="30"/>
+            <source>Login</source>
+            <translation>Autentificare</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="31"/>
+            <source>Login failed</source>
+            <translation>Autentificare eșuată</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="32"/>
+            <source>Login succeeded</source>
+            <translation>Autentificare cu succes</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="33"/>
+            <source>Password</source>
+            <translation>Parolă</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="34"/>
+            <source>Please enter a password!</source>
+            <translation>Introduceți o parolă!</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="35"/>
+            <source>Change password</source>
+            <translation>Schimbă parola</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="36"/>
+            <source>Enter your username and password</source>
+            <translation>Introduceți numele de utilizator și parola</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="38"/>
+            <source>Enter your username</source>
+            <translation>Introduceți numele de utilizator</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="39"/>
+            <source>Enter your password</source>
+            <translation>Introduceți parola</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="40"/>
+            <source>Password:</source>
+            <translation>Parolă:</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="41"/>
+            <source>Show password</source>
+            <translation>Arată parola</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="42"/>
+            <source>Hide password</source>
+            <translation>Ascunde parola</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="43"/>
+            <source>Reboot</source>
+            <translation>Repornește</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="44"/>
+            <source>Session</source>
+            <translation>Sesiune</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="45"/>
+            <source>Shutdown</source>
+            <translation>Oprește</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="46"/>
+            <source>Suspend</source>
+            <translation>Suspendă</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="47"/>
+            <source>Hibernate</source>
+            <translation>Hibernează</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="48"/>
+            <source>Username</source>
+            <translation>Nume de utilizator</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="50"/>
+            <source>Password change aborted because maximum tries reached</source>
+            <translation>Schimbarea parolei a fost abandonată pentru că a fost atins numărul maxim de încercări</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="51"/>
+            <source>New password change round! Please input current password again!</source>
+            <translation>Rundă nouă pentru schimbarea parolei! Introduceți din nou parola actuală!</translation>
+        </message>
+        <message>
+            <location filename="../../components/common/TextConstants.qml" line="37"/>
+            <source>Select your user and enter password</source>
+            <translation>Alegeți-vă numele și introduceți parola</translation>
+        </message>
+    </context>
 </TS>
