--- sddm-0.17.0/src/daemon/XorgDisplayServer.cpp.omv~	2018-02-22 00:20:05.457303000 +0100
+++ sddm-0.17.0/src/daemon/XorgDisplayServer.cpp	2018-02-22 00:20:14.470326606 +0100
@@ -103,7 +103,7 @@ namespace SDDM {
 
         // set up the auth entry
         char cookieName[] = "MIT-MAGIC-COOKIE-1";
-        char displayNumber[m_display.size() + 1] = { 0 };
+        char displayNumber[m_display.size() + 1];
         strcpy(displayNumber, qPrintable(m_display.mid(1))); // Need to skip the ':'
         auth.family = FamilyLocal;
         auth.address = localhost;
--- sddm-0.17.0/src/helper/UserSession.cpp.omv~	2018-02-22 00:18:54.674113558 +0100
+++ sddm-0.17.0/src/helper/UserSession.cpp	2018-02-22 00:19:43.910246109 +0100
@@ -188,7 +188,7 @@ namespace SDDM {
 
             // set up the auth entry
             char cookieName[] = "MIT-MAGIC-COOKIE-1";
-            char displayNumber[display.size() + 1] = { 0 };
+            char displayNumber[display.size() + 1];
             strcpy(displayNumber, qPrintable(display.mid(1))); // Need to skip the ':'
             auth.family = FamilyLocal;
             auth.address = localhost;
