diff -Naur sddm-0.14.0-20151218/CMakeLists.txt sddm-0.14.0-20151218.tpg/CMakeLists.txt
--- sddm-0.14.0-20151218/CMakeLists.txt	2015-12-09 13:24:52.000000000 +0000
+++ sddm-0.14.0-20151218.tpg/CMakeLists.txt	2016-01-09 19:46:12.091819343 +0000
@@ -90,6 +90,20 @@
 find_package(Qt5DBus REQUIRED)
 find_package(Qt5LinguistTools REQUIRED)
 
+# AccountsService Qt5 binding
+find_package(QtAccountsService)
+set_package_properties(QtAccountsService PROPERTIES 
+    DESCRIPTION "Qt-style API to use freedesktop.org's AccountsService DBus service"
+    URL "https://github.com/hawaii-desktop/qtaccountsservice"
+    TYPE OPTIONAL
+    PURPOSE "Use AccountsService's iconFileName"
+)
+set(HAVE_QTACCOUNTSSERVICE FALSE)
+if (QtAccountsService_FOUND)
+    add_definitions(-DHAVE_QTACCOUNTSSERVICE)
+    set(HAVE_QTACCOUNTSSERVICE TRUE)
+endif()
+
 # find qt5 imports dir
 get_target_property(QMAKE_EXECUTABLE Qt5::qmake LOCATION)
 if(NOT QT_IMPORTS_DIR)
diff -Naur sddm-0.14.0-20151218/src/greeter/CMakeLists.txt sddm-0.14.0-20151218.tpg/src/greeter/CMakeLists.txt
--- sddm-0.14.0-20151218/src/greeter/CMakeLists.txt	2015-12-09 13:24:52.000000000 +0000
+++ sddm-0.14.0-20151218.tpg/src/greeter/CMakeLists.txt	2016-01-09 19:48:05.123840371 +0000
@@ -26,6 +26,10 @@
     target_link_libraries(sddm-greeter ${JOURNALD_LIBRARIES})
 endif()
 
+if(HAVE_QTACCOUNTSSERVICE)
+    target_link_libraries(sddm-greeter QtAccountsService::QtAccountsService)
+endif()
+
 # Translations
 add_dependencies(sddm-greeter components-translation)
 add_dependencies(sddm-greeter themes-translation)
diff -Naur sddm-0.14.0-20151218/src/greeter/UserModel.cpp sddm-0.14.0-20151218.tpg/src/greeter/UserModel.cpp
--- sddm-0.14.0-20151218/src/greeter/UserModel.cpp	2015-12-09 13:24:52.000000000 +0000
+++ sddm-0.14.0-20151218.tpg/src/greeter/UserModel.cpp	2016-01-09 19:55:11.292226025 +0000
@@ -1,5 +1,7 @@
 /***************************************************************************
 * Copyright (c) 2013 Abdurrahman AVCI <abdurrahmanavci@gmail.com>
+* Copyright (c) 2015 Leslie Zhai <xiang.zhai@i-soft.com.cn>
+* Copyright (c) 2015 Mingye Wang (Arthur2e5) <arthur200126@gmail.com>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
@@ -55,6 +57,9 @@
         const QString defaultFace = QStringLiteral("%1/.face.icon").arg(facesDir);
 
         struct passwd *current_pw;
+#if HAVE_QTACCOUNTSSERVICE
+        asAccountsManager = new QtAccountsService::AccountsManager;
+#endif
         while ((current_pw = getpwent()) != nullptr) {
 
             // skip entries with uids smaller than minimum uid
@@ -79,6 +84,9 @@
             user->homeDir = QString::fromLocal8Bit(current_pw->pw_dir);
             user->uid = int(current_pw->pw_uid);
             user->gid = int(current_pw->pw_gid);
+#if HAVE_QTACCOUNTSSERVICE
+            QtAccountsService::UserAccount *asUserAccount = asAccountsManager->findUserByName(user->name);
+#endif
             // if shadow is used pw_passwd will be 'x' nevertheless, so this
             // will always be true
             user->needsPassword = strcmp(current_pw->pw_passwd, "") != 0;
@@ -112,6 +120,10 @@
 
                 if (QFile::exists(userFace))
                     user->icon = userFace;
+#if HAVE_QTACCOUNTSSERVICE
+		else if (asUserAccount && QFile::exists(asUserAccount->iconFileName()))
+		    user->icon = asUserAccount->iconFileName();	// accountservice user face
+#endif
                 else if (QFile::exists(systemFace))
                     user->icon = systemFace;
             }
@@ -120,6 +132,12 @@
 
     UserModel::~UserModel() {
         delete d;
+#if HAVE_QTACCOUNTSSERVICE
+        if (asAccountsManager) {
+            delete asAccountsManager;
+            asAccountsManager = nullptr;
+        }
+#endif
     }
 
     QHash<int, QByteArray> UserModel::roleNames() const {
diff -Naur sddm-0.14.0-20151218/src/greeter/UserModel.h sddm-0.14.0-20151218.tpg/src/greeter/UserModel.h
--- sddm-0.14.0-20151218/src/greeter/UserModel.h	2015-12-09 13:24:52.000000000 +0000
+++ sddm-0.14.0-20151218.tpg/src/greeter/UserModel.h	2016-01-09 19:54:06.987508490 +0000
@@ -1,5 +1,7 @@
 /***************************************************************************
 * Copyright (c) 2013 Abdurrahman AVCI <abdurrahmanavci@gmail.com>
+* Copyright (c) 2015 Leslie Zhai <xiang.zhai@i-soft.com.cn>
+* Copyright (c) 2015 Mingye Wang (Arthur2e5) <arthur200126@gmail.com>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
@@ -23,6 +25,9 @@
 #include <QAbstractListModel>
 
 #include <QHash>
+#if HAVE_QTACCOUNTSSERVICE
+#include <QtAccountsService/AccountsManager>  /* QtAccountsService */
+#endif
 
 namespace SDDM {
     class UserModelPrivate;
@@ -57,6 +62,9 @@
         int disableAvatarsThreshold() const;
     private:
         UserModelPrivate *d { nullptr };
+#if HAVE_QTACCOUNTSSERVICE
+        QtAccountsService::AccountsManager *asAccountsManager { nullptr };
+#endif
     };
 }
 
