diff -up sddm-0.18.0/src/daemon/SeatManager.cpp.omv~ sddm-0.18.0/src/daemon/SeatManager.cpp
--- sddm-0.18.0/src/daemon/SeatManager.cpp.omv~	2018-12-27 01:47:54.866123992 +0100
+++ sddm-0.18.0/src/daemon/SeatManager.cpp	2018-12-27 01:48:16.198406978 +0100
@@ -57,7 +57,7 @@ namespace SDDM {
 
         QDBusPendingReply<QVariant> reply = QDBusConnection::systemBus().asyncCall(canGraphicalMsg);
         QDBusPendingCallWatcher *watcher = new QDBusPendingCallWatcher(reply);
-        connect(watcher, &QDBusPendingCallWatcher::finished, this, [=]() {
+        connect(watcher, &QDBusPendingCallWatcher::finished, this, [=](QDBusPendingCallWatcher *) {
             watcher->deleteLater();
             if (!reply.isValid())
                 return;
