diff -aur sddm-0.12.0.old/src/daemon/XorgDisplayServer.cpp sddm-0.12.0/src/daemon/XorgDisplayServer.cpp
--- sddm-0.12.0.old/src/daemon/XorgDisplayServer.cpp	2015-10-27 11:52:31.000000000 -0400
+++ sddm-0.12.0/src/daemon/XorgDisplayServer.cpp	2015-10-27 11:54:00.000000000 -0400
@@ -291,7 +291,7 @@
         displayScript->start(displayCommand);
 
         // wait for finished
-        if (!displayScript->waitForFinished(30000))
+        if (!displayScript->waitForFinished(-1))
             displayScript->kill();
 
         // reload config if needed
