diff -Naur sddm-0.17.0/src/daemon/XorgDisplayServer.cpp sddm-0.17.0.tpg/src/daemon/XorgDisplayServer.cpp
--- sddm-0.17.0/src/daemon/XorgDisplayServer.cpp	2017-12-05 12:00:22.010667376 +0000
+++ sddm-0.17.0.tpg/src/daemon/XorgDisplayServer.cpp	2017-12-11 18:32:28.339121658 +0000
@@ -316,7 +316,7 @@
         connect(displayScript, SIGNAL(finished(int,QProcess::ExitStatus)), displayScript, SLOT(deleteLater()));
 
         // wait for finished
-        if (!displayScript->waitForFinished(30000))
+        if (!displayScript->waitForFinished(-1))
             displayScript->kill();
 
         // reload config if needed
