--- sddm-20131021/data/sddm.conf.in.omv~	2013-10-23 01:43:48.579923302 +0200
+++ sddm-20131021/data/sddm.conf.in	2013-10-23 01:52:37.284313587 +0200
@@ -21,7 +21,7 @@ RebootCommand=@REBOOT_COMMAND@
 
 # Path of the directory containing session files,
 # e.g kde-plasma.desktop
-SessionsDir=/usr/share/xsessions
+SessionsDir=/usr/share/apps/kdm/sessions
 
 # Name of the session file of the last session
 # selected. This session will be preselected when
@@ -49,7 +49,7 @@ CurrentTheme=maui
 
 # Minimum user id of the users to be listed in the
 # user interface
-MinimumUid=1000
+MinimumUid=500
 
 # Maximum user id of the users to be listed in the
 # user interface
