diff -Naur sddm-0.18.1-20191216/services/sddm.service.in sddm-0.18.1-20191216.tpg/services/sddm.service.in
--- sddm-0.18.1-20191216/services/sddm.service.in	2019-12-16 12:16:35.901032635 +0000
+++ sddm-0.18.1-20191216.tpg/services/sddm.service.in	2019-12-16 12:16:17.934032529 +0000
@@ -7,6 +7,7 @@
 StartLimitBurst=2
 
 [Service]
+ExecStartPre=-/bin/plymouth quit --retain-splash
 ExecStart=@CMAKE_INSTALL_FULL_BINDIR@/sddm
 Restart=always
 
