diff -Naur sddm-0.11.0/src/daemon/XorgDisplayServer.cpp sddm-0.11.0.tpg/src/daemon/XorgDisplayServer.cpp
--- sddm-0.11.0/src/daemon/XorgDisplayServer.cpp	2015-04-08 21:34:38.575063425 +0000
+++ sddm-0.11.0.tpg/src/daemon/XorgDisplayServer.cpp	2015-04-08 21:34:06.800215170 +0000
@@ -239,6 +239,8 @@
         qDebug() << "Running display setup script " << displayCommand;
         displayScript->start(displayCommand);
         displayScript->waitForFinished(-1);
+        //reload config if needed
+        mainConfig.load();
     }
 
     void XorgDisplayServer::changeOwner(const QString &fileName) {
