diff -Naur sddm-0.16.0/src/common/Configuration.h sddm-0.16.0.tpg/src/common/Configuration.h
--- sddm-0.16.0/src/common/Configuration.h	2017-10-04 17:13:38.033912912 +0000
+++ sddm-0.16.0.tpg/src/common/Configuration.h	2017-10-17 16:29:18.997786346 +0000
@@ -101,7 +101,7 @@
 
     Config(StateConfig, []()->QString{auto tmp = getpwnam("sddm"); return tmp ? QString::fromLocal8Bit(tmp->pw_dir) : QStringLiteral(STATE_DIR);}().append(QStringLiteral("/state.conf")), QString(), QString(),
         Section(Last,
-            Entry(Session,         QString,     QString(),                                      _S("Name of the session for the last logged-in user.\n"
+            Entry(Session,         QString,     _S("/usr/share/wayland-sessions/plasmawayland.desktop"),      _S("Name of the session for the last logged-in user.\n"
                                                                                                    "This session will be preselected when the login screen appears."));
             Entry(User,            QString,     QString(),                                      _S("Name of the last logged-in user.\n"
                                                                                                    "This user will be preselected when the login screen appears"));
