diff -urN sddm-0.11.0/src/daemon/Display.cpp sddm-0.11.0-patched/src/daemon/Display.cpp
--- sddm-0.11.0/src/daemon/Display.cpp	2014-11-21 01:47:24.000000000 +1000
+++ sddm-0.11.0-patched/src/daemon/Display.cpp	2015-05-06 13:41:53.376997954 +1000
@@ -271,7 +271,22 @@
         // last session later, in slotAuthenticationFinished()
         m_sessionName = sessionFileName;
 
+        QProcessEnvironment sysenv = QProcessEnvironment::systemEnvironment();
         QProcessEnvironment env;
+        env.insert("LANG", sysenv.value("LANG"));
+        env.insert("LANGUAGE", sysenv.value("LANGUAGE"));
+        env.insert("LC_CTYPE", sysenv.value("LC_CTYPE"));
+        env.insert("LC_NUMERIC", sysenv.value("LC_NUMERIC"));
+        env.insert("LC_TIME", sysenv.value("LC_TIME"));
+        env.insert("LC_COLLATE", sysenv.value("LC_COLLATE"));
+        env.insert("LC_MONETARY", sysenv.value("LC_MONETARY"));
+        env.insert("LC_MESSAGES", sysenv.value("LC_MESSAGES"));
+        env.insert("LC_PAPER", sysenv.value("LC_PAPER"));
+        env.insert("LC_NAME", sysenv.value("LC_NAME"));
+        env.insert("LC_ADDRESS", sysenv.value("LC_ADDRESS"));
+        env.insert("LC_TELEPHONE", sysenv.value("LC_TELEPHONE"));
+        env.insert("LC_MEASUREMENT", sysenv.value("LC_MEASUREMENT"));
+        env.insert("LC_IDENTIFICATION", sysenv.value("LC_IDENTIFICATION"));
         env.insert("PATH", mainConfig.Users.DefaultPath.get());
         env.insert("DISPLAY", name());
         env.insert("XDG_SEAT", seat()->name());
