diff -Naur sddm-0.18.1-20191216/services/sddm.service.in sddm-0.18.1-20191216.tpg/services/sddm.service.in
--- sddm-0.18.1-20191216/services/sddm.service.in	2019-12-16 12:53:36.334045723 +0000
+++ sddm-0.18.1-20191216.tpg/services/sddm.service.in	2019-12-16 12:54:34.891046069 +0000
@@ -10,6 +10,7 @@
 ExecStartPre=-/bin/plymouth quit --retain-splash
 ExecStart=@CMAKE_INSTALL_FULL_BINDIR@/sddm
 Restart=always
+EnvironmentFile=-/etc/sysconfig/sddm
 
 [Install]
 Alias=display-manager.service
