diff -Naur sddm-0.14.0-20151218/CMakeLists.txt sddm-0.14.0-20151218.tpg/CMakeLists.txt
--- sddm-0.14.0-20151218/CMakeLists.txt	2016-01-09 21:42:03.297667263 +0000
+++ sddm-0.14.0-20151218.tpg/CMakeLists.txt	2016-01-09 21:43:54.901608034 +0000
@@ -104,6 +104,11 @@
     set(HAVE_QTACCOUNTSSERVICE TRUE)
 endif()
 
+if (ENABLE_PLYMOUTH)
+    add_definitions(-DHAVE_PLYMOUTH)
+    set(HAVE_PLYMOUTH TRUE)
+endif()
+
 # find qt5 imports dir
 get_target_property(QMAKE_EXECUTABLE Qt5::qmake LOCATION)
 if(NOT QT_IMPORTS_DIR)
diff -Naur sddm-0.14.0-20151218/services/sddm.service.in sddm-0.14.0-20151218.tpg/services/sddm.service.in
--- sddm-0.14.0-20151218/services/sddm.service.in	2015-12-09 13:24:52.000000000 +0000
+++ sddm-0.14.0-20151218.tpg/services/sddm.service.in	2016-01-09 21:44:27.526466664 +0000
@@ -1,8 +1,11 @@
 [Unit]
 Description=Simple Desktop Display Manager
 Documentation=man:sddm(1) man:sddm.conf(5)
-Conflicts=getty@tty1.service
-After=systemd-user-sessions.service getty@tty1.service plymouth-quit.service
+Conflicts=getty@tty1.service plymouth-quit.service
+After=getty@tty1.service plymouth-quit.service
+After=rc-local.service systemd-user-sessions.service
+
+OnFailure=plymouth-quit.service
 
 [Service]
 ExecStart=@CMAKE_INSTALL_FULL_BINDIR@/sddm
diff -Naur sddm-0.14.0-20151218/src/daemon/DisplayManager.cpp sddm-0.14.0-20151218.tpg/src/daemon/DisplayManager.cpp
--- sddm-0.14.0-20151218/src/daemon/DisplayManager.cpp	2015-12-09 13:24:52.000000000 +0000
+++ sddm-0.14.0-20151218.tpg/src/daemon/DisplayManager.cpp	2016-01-09 21:44:27.527466690 +0000
@@ -40,6 +40,9 @@
         QDBusConnection connection = (daemonApp->testing()) ? QDBusConnection::sessionBus() : QDBusConnection::systemBus();
         connection.registerService(DISPLAYMANAGER_SERVICE);
         connection.registerObject(DISPLAYMANAGER_PATH, this);
+#if HAVE_PLYMOUTH
+        system("/usr/bin/plymouth deactivate");
+#endif
     }
 
     QString DisplayManager::seatPath(const QString &seatName) {
diff -Naur sddm-0.14.0-20151218/src/daemon/Greeter.cpp sddm-0.14.0-20151218.tpg/src/daemon/Greeter.cpp
--- sddm-0.14.0-20151218/src/daemon/Greeter.cpp	2015-12-09 13:24:52.000000000 +0000
+++ sddm-0.14.0-20151218.tpg/src/daemon/Greeter.cpp	2016-01-09 21:46:25.494571619 +0000
@@ -71,6 +71,11 @@
             // log message
             qDebug() << "Greeter starting...";
 
+#if HAVE_PLYMOUTH
+	    qDebug() << "Plymouth quit...";
+            system("/usr/bin/plymouth --wait quit --retain-splash");
+#endif
+
             // set process environment
             QProcessEnvironment env = QProcessEnvironment::systemEnvironment();
             env.insert(QStringLiteral("DISPLAY"), m_display->name());
